components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/AuthorizationHandlerManager.java

```diff
@@ -208,6 +213,205 @@ private OAuth2AuthorizeRespDTO validateAuthzRequest(OAuth2AuthorizeReqDTO authzR
+        } finally {
+            // clears authorization request context
+            OAuth2Util.clearAuthzRequestContext();
+        }
+        return authorizeRespDTO;
+    }
+
+
+    /**
+     * validated requested scopes.
+     *
+     * @param authzReqMsgCtx authzReqMsgCtx
+     * @param authzHandler ResponseTypeHandler
+     */
+    private void validateRequestedScopes(OAuthAuthzReqMessageContext authzReqMsgCtx,
+                                            ResponseTypeHandler authzHandler) throws IdentityOAuth2Exception,
+            IdentityOAuth2ScopeValidationException {
+
+        // Get OIDC scopes from requested scopes and drop from requested scopes
+        String[] requestedOIDCScopes = null;
+        try {
+            requestedOIDCScopes = OAuth2Util.getRequestedOIDCScopes(authzReqMsgCtx.getAuthorizationReqDTO()
+                    .getScopes());
+            String[] oidcRemovedScopes  = OAuth2Util.removeOIDCScopesFromRequestedScopes(authzReqMsgCtx
+                    .getAuthorizationReqDTO().getScopes());
+            authzReqMsgCtx.getAuthorizationReqDTO().setScopes(oidcRemovedScopes);
+
+        } catch (IdentityOAuthAdminException e) {
+            throw new IdentityOAuth2Exception(e.getMessage());
+        }
+
+        // Get scopes that specified in the allowed scopes list.
+        List<String> requestedAllowedScopes = getAllowedScopesFromRequestedScopes(authzReqMsgCtx);
+
+        // Validate internal scopes.
+        String[] authorizedInternalScopes = new String[0];
+        boolean isManagementApp = isManagementApp(authzReqMsgCtx.getAuthorizationReqDTO());
+        if (isManagementApp) {
+            if (log.isDebugEnabled()) {
+                log.debug("Handling the internal scope validation.");
+            }
+            authorizedInternalScopes = getAuthorizedInternalScopes(authzReqMsgCtx);
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("Skipping the internal scope validation as the application is not" +
+                        " configured as Management App");
+            }
+        }
+
+        // Remove the internal scopes from requested scopes for further validation.
+        removeInternalScopesFromRequestedScopes(authzReqMsgCtx);
+
+        // Adding the authorized internal scopes to tokReqMsgCtx for any special validators to use.
+        authzReqMsgCtx.setAuthorizedInternalScopes(authorizedInternalScopes);
+
+        boolean isDropUnregisteredScopes = OAuthServerConfiguration.getInstance().isDropUnregisteredScopes();
+        if (isDropUnregisteredScopes) {
+            if (log.isDebugEnabled()) {
+                log.debug("DropUnregisteredScopes config is enabled. Attempting to drop unregistered scopes.");
+            }
+            dropUnregisteredScopeFromRequestedScopes(authzReqMsgCtx);
+        }
+
+        //validate scopes using global scope validators
+        boolean valid = validateScopes(authzReqMsgCtx, authzHandler);
```

Comment:
variable name and the method name needs to be improved. Let's use names that are suggestive of a boolean